charmcraft.yaml

```diff
@@ -68,6 +68,28 @@ config:
       description: "Whether or not csi-*plugin-provisioner deployments use host-networking"
       type: boolean
 
+    csidriver-name-formatter:
+      default: "{name}"
+      description: |
+        Formatter for the 2 CSIDrivers managed by this charm:
+          - rbd.csi.ceph.com
+          - cephfs.csi.ceph.com
+
+        The formatter is a string that can contain the following placeholders:
+          - {name}             - original name from the manifest
+          - {app}              - the name of the juju application
+          - {namespace}        - the charm configured namespace
+
+        Example:
+          juju config ceph-csi csidriver-name-formatter="{name}.{app}"
+
+        The default is to use the built-in storage classname
+
+        NOTE: Can only be specified on deployment since some
+        attributes of kubernetes resources are non-modifiable.
+        The admin is responsible for creating the namespace.
```

Comment:
The `admin is responsible...` part is a copy/past error.  The rest of the note could be improved